File: 118. Pascal's Triangle.go
func generate(numRows int) [][]int {
	var res [][]int;

	for i := 1; i <= numRows; i++{
		l := make([]int, i);
		for j := range l{
			l[j] = 1;
		}
		res = append(res, l);
	}

	for i := 2; i < numRows; i++{
		for j := 1; j < len(res[i]) - 1; j++{
			res [i][j] = res[i - 1][j - 1] + res[i - 1][j];
		}
	}

	return res;
}

File: 118. Pascal's Triangle.java

import java.security.DrbgParameters;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();

        for(int i = 1; i <= numRows; ++i){
            Integer[] l = new Integer[i];
            Arrays.fill(l,  1);
            res.add(Arrays.asList(l));
        }

        for (int i = 2; i < numRows; ++i)
            for (int j = 1; j < res.get(i).size() - 1; ++j)
                res.get(i).set(j, res.get(i - 1).get(j - 1) + res.get(i - 1).get(j));
        return res;
    }
}